main.go

```diff
@@ -32,85 +34,139 @@ type MyResponse struct {
-  bucketName := event.Records[0].S3.Bucket.Name
-  objectKey := event.Records[0].S3.Object.Key
+type LambdaTagValues struct {
+	pass string
+	fail string
+}
 
-	sess := session.Must(session.NewSession())
+type DownloaderInterface interface {
+	Download(w io.WriterAt, input *s3.GetObjectInput, options ...func(*s3manager.Downloader)) (n int64, err error)
+}
 
-	endpoint := os.Getenv("AWS_S3_ENDPOINT")
-	sess.Config.Endpoint = &endpoint
-	sess.Config.S3ForcePathStyle = aws.Bool(true)
+type Lambda struct {
+	tagKey     string
+	tagValues  LambdaTagValues
+	s3         s3iface.S3API
+	downloader DownloaderInterface
+}
 
-	downloader := s3manager.NewDownloader(sess)
+func (l Lambda) path() string {
+	return "/tmp/file"
+}
+
+func (l Lambda) downloadFile(bucket string, key string) error {
+	filename := l.path()
 
-	filename := "/tmp/file"
 	f, err := os.Create(filename)
 	if err != nil {
 		log.Printf("failed to create file %q, %v", filename, err)
-		return MyResponse{}, fmt.Errorf("failed to create file %q, %v", filename, err)
+		return fmt.Errorf("failed to create file %q, %v", filename, err)
 	}
 
-	_, err = downloader.Download(f, &s3.GetObjectInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(objectKey),
+	_, err = l.downloader.Download(f, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 	})
 
 	if err != nil {
 		log.Printf("failed to download file, %v", err)
-		return MyResponse{}, fmt.Errorf("failed to download file, %v", err)
+		return fmt.Errorf("failed to download file, %v", err)
 	}
 
-	// scan with clamav
-	status := "ok"
-	cmd := exec.Command("./bin/clamscan", filename)
+	return nil
+}
+
+func (l Lambda) scanFile() (string, error) {
+	status := l.tagValues.pass
+	cmd := exec.Command("./bin/clamscan", l.path())
 	if err := cmd.Run(); err != nil {
 		if exitError, ok := err.(*exec.ExitError); ok {
 			if exitError.ExitCode() == 1 {
-				status = "infected"
+				status = l.tagValues.fail
 			} else {
 				log.Printf("failed to scan file, %v", err)
-				return MyResponse{}, fmt.Errorf("failed to scan file, %v", err)
+				return "", fmt.Errorf("failed to scan file, %v", err)
 			}
 		} else {
 			log.Printf("failed to scan file, %v", err)
-			return MyResponse{}, fmt.Errorf("failed to scan file, %v", err)
+			return "", fmt.Errorf("failed to scan file, %v", err)
 		}
 	}
 
-	// tag document
-	s3client := s3.New(sess)
+	return status, nil
+}
 
-	tagging, err := s3client.GetObjectTagging(&s3.GetObjectTaggingInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(objectKey),
+func (l Lambda) tagFile(bucket string, key string, status string) error {
+	tagging, err := l.s3.GetObjectTagging(&s3.GetObjectTaggingInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 	})
 
 	if err != nil {
 		log.Printf("failed to get tags, %v", err)
-		return MyResponse{}, fmt.Errorf("failed to get tags, %v", err)
+		return fmt.Errorf("failed to get tags, %v", err)
```

Comment:
you can use `%w` for errors, then they can be unwrapped

Reply:
and you could move the logging up to `HandleEvent` from these methods so it isn't duplicated as much